scanner/keppel/processor/processor.go

```diff
@@ -112,38 +112,54 @@ func (p *Processor) ProcessReport(report models.TrivyReport, componentVersionId
 	}
 }
 
-func (p *Processor) GetComponent(name string) (*client.Component, error) {
-	r, err := client.ListComponents(context.Background(), *p.Client, &client.ComponentFilter{
-		ComponentName: []string{name},
-	}, 1)
+// GetAllComponents will fetch all available Components using pagination.
+// pageSize specifies how many object we want to fetch in a row. Then we use cursor
+// to fetch the next batch.
+func (p *Processor) GetAllComponents(filter *client.ComponentFilter, pageSize int) ([]*client.Component, error) {
+	var allComponents []*client.Component
+	cursor := "0" // Set initial cursor to "0"
+
+	for {
+		listComponentsResp, err := client.ListComponents(context.Background(), *p.Client, filter, pageSize, cursor)
+		if err != nil {
+			return nil, fmt.Errorf("cannot list Components: %w", err)
+		}
 
-	if err != nil {
-		return nil, err
-	}
+		if len(listComponentsResp.Components.Edges) == 0 {
+			break
+		}
 
-	var component *client.Component
-	if len(r.Components.Edges) > 0 {
-		component = r.Components.Edges[0].GetNode()
-	}
+		for _, edge := range listComponentsResp.Components.Edges {
+			allComponents = append(allComponents, edge.Node)
+		}
 
-	return component, nil
+		if len(listComponentsResp.Components.Edges) < pageSize {
+			break
+		}
+
+		// Update cursor for the next iteration
+		cursor = listComponentsResp.Components.Edges[len(listComponentsResp.Components.Edges)-1].Cursor
+	}
+	return allComponents, nil
 }
```

Comment:
See my comment above, why not getting version + component at once and build a list of concrete manifests out of that?

Reply:
Implemented in [33b65ed](https://github.com/cloudoperators/heureka/pull/309/commits/33b65edb68388ba945d2038d6ce25802756bfb32)